### PR TITLE
Ecc 679

### PR DESCRIPTION
**Pull request by NandanaRaghu**

### CODE REVIEW

File: test/config/ErrorHandlerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      val result = errorHandler.onClientError(mockRequest, statusCode = INTERNAL_SERVER_ERROR)
+      val page   = RegistrationPage(contentAsString(result))
+      status(result) shouldBe INTERNAL_SERVER_ERROR
+      page.title should startWith("Something went wrong. Please try again later.")
+    }
+
+    /*
+
+
+        "Redirect to the notfound page on 404 error with InvalidPathParameter" in {
```
Comment (yasaracar) [inline test/config/ErrorHandlerSpec.scala:76]: Commented out code

**Review by yasaracar — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/xieoricommoncomponentfrontend/config/ErrorHandler.scala [hunk 2]
```diff
@@ -55,4 +57,26 @@ class ErrorHandler @Inject() (
     }
   }
 
+  override def onServerError(request: RequestHeader, exception: Throwable): Future[Result] = {
+
+    implicit val req: Request[_] = Request(request, "")
+
+    exception match {
+      case sessionTimeOut: SessionTimeOutException =>
+        // $COVERAGE-OFF$Loggers
```
[the remark this answers is not in the document]
Comment (NandanaRaghu) [reply app/uk/gov/hmrc/xieoricommoncomponentfrontend/config/ErrorHandler.scala]: removed coverage statement

File: app/uk/gov/hmrc/xieoricommoncomponentfrontend/controllers/ConfirmDetailsController.scala [hunk 3]
```diff
@@ -83,15 +73,19 @@ class ConfirmDetailsController @Inject() (
...
-                )
-              case Left(_) => InternalServerError(errorTemplateView())
+            groupEnrolment.getEori(loggedInUser).flatMap {
+              case Some(gbEori) =>
+                subscriptionDisplayService.getSubscriptionDisplay(gbEori).map {
+                  case Right(response) =>
+                    BadRequest(
+                      confirmDetailsView(
+                        formWithErrors,
+                        ConfirmDetailsViewModel(response, loggedInUser.affinityGroup.get)
```
[the remark this answers is not in the document]
Comment (NandanaRaghu) [reply app/uk/gov/hmrc/xieoricommoncomponentfrontend/controllers/ConfirmDetailsController.scala]: fixed the code and removed get

File: test/config/ErrorHandlerSpec.scala (hunk 1 above)
Comment (NandanaRaghu) [reply test/config/ErrorHandlerSpec.scala]: removed commented code

**Review by yasaracar — APPROVED** [review]

**Pull request merged**